### PR TITLE
feat(subtitle-button):added new aria label for subtitle button [PRO-411]

### PR DESCRIPTION
**Pull request by AbrakaBarna**
Fixes: https://twentythreecom.atlassian.net/browse/PRO-411

### CODE REVIEW

File: src/subtitle-button/subtitle-button.liquid [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
-    <button class="subtitle-button{% if subtitleLocale != empty %} subtitle-button-active{% endif %}" aria-controls="subtitleOptions" aria-label="{{"closed_captioning"|translate}}" label="{{"closed_captioning"|translate}}" tabindex="1050">
-      
+    <button
+      class="subtitle-button{% if subtitleLocale != empty %} subtitle-button-active{% endif %}"
+      aria-controls="subtitleOptions"
+      aria-label="{{initialSubtitleAriaLabel|translate}}"
+      label="{{"initialSubtitleAriaLabel"|translate}}"
+      tabindex="1050">
+
       <span class="hide-visually">
```
Comment (AbrakaBarna) [inline src/subtitle-button/subtitle-button.liquid:10]: I would have a question for this. When are we using this hide-visually span?

Comment (steffentchr) [reply src/subtitle-button/subtitle-button.liquid:10]: I guess the purpose is to have button text for the browser to index, but not to show -- and guess it's down to labeling stuff pre aria.

File: src/accessibility/accessibility.js [hunk 2]
```diff
@@ -52,6 +58,12 @@ Player.provide('accessibility',
                     Player.set('qualityMenuExpanded', true)
                     $(document.activeElement).attr("aria-label", Player.translate("expanded"))
                   }
+                  if($(document.activeElement).hasClass('subtitle-button')){
+                    Player.set('subtitleMenuExpanded', true)
+                    var subtitleOn = Player.get('subtitleLocale') != "";
+                    if(subtitleOn) $(document.activeElement).attr("aria-label", Player.translate("subtitle_on_expanded"))
+                    else $(document.activeElement).attr("aria-label", Player.translate("subtitle_off_expanded"))
```
Comment (steffentchr) [inline src/accessibility/accessibility.js:65]: I guess this is equivalent:

```suggestion
    $(document.activeElement).attr("aria-label", Player.translate( Player.get('subtitleLocale') != "" ? "subtitle_on_expanded" : "subtitle_off_expanded"))
```

File: src/subtitle-button/subtitle-button.js [hunk 3]
```diff
@@ -50,3 +59,16 @@ Player.translate("disable_closed_captioning",{
 Player.translate("none",{
     en: "None"
 });
+
+Player.translate("subtitle_on_expanded",{
+  en: "Captions are turned on. Expanded subtitle popup button. Use tab to navigate between the options. Press enter key to collapse."
```
Comment (steffentchr) [inline src/subtitle-button/subtitle-button.js:64]: We should make an effort separately to translate these to our covered languages.

**Review by steffentchr — APPROVED** [review]

File: src/subtitle-button/subtitle-button.liquid (hunk 1 above)
Comment (AbrakaBarna) [reply src/subtitle-button/subtitle-button.liquid:10]: alright, we need it in this case. thanks for the answer

File: src/subtitle-button/subtitle-button.js (hunk 3 above)
Comment (AbrakaBarna) [reply src/subtitle-button/subtitle-button.js:64]: should I create a ticket for it?

**Pull request merged**